src/group_theory/perm/basic.lean

```diff
@@ -159,6 +159,23 @@ begin
   simpa using equiv.congr_fun h ⟨a, b⟩,
 end
 
+/-- `equiv.perm.subtype_congr` as a `monoid_hom`. -/
+@[simps] def subtype_congr_hom (p : α → Prop) [decidable_pred p] :
+  (perm {a // p a}) × (perm {a // ¬ p a}) →* perm α :=
+{ to_fun := λ pair, perm.subtype_congr pair.fst pair.snd,
+  map_one' := perm.subtype_congr.refl,
+  map_mul' := λ _ _, (perm.subtype_congr.trans _ _ _ _).symm }
+
+lemma subtype_congr_hom_injective (p : α → Prop) [decidable_pred p] :
```

Comment:
Note that the only reason I added this lemma for `sigma_congr` above was to prove

https://github.com/leanprover-community/mathlib/blob/49517d903fedea734618349ce85de26131b24e7a/src/group_theory/perm/subgroup.lean#L28-L32

Up to you whether you want to try and prove a similar statement in this PR.

Reply:
For my now-added lemma, it seems to me that the `fintype` are only satisfied iff `[fintype α]`. Should I phrase it so?

Reply:
No, because then your lemma will not match other fintype instances

Reply:
How do you mean? What's a way to have `[fintype (perm {a // p a} × perm {a // ¬ p a})]` without `[fintype α]`?

Reply:
You're correct that if there's an iff relation, but that doesn't mean that the "obvious" construction is the one present in the term the simplifier is matching against - this has the same trappings as decidable, and just happens to appear much less often.